tests/ui/needless_type_cast.fixed

```diff
@@ -0,0 +1,55 @@
+#![warn(clippy::needless_type_cast)]
+#![allow(clippy::no_effect, clippy::unnecessary_cast, unused)]
+
+fn main() {
+    // Should lint: let binding always cast to i32
```

Comment:
Those tests only cover local variables and a few binary ops. What about function and method calls? Especially if those functions might be generic, and doubly especially if they have a generic return type, like `.sum()`.